theia-extensions/viewer-prototype/src/node/trace-server-url-provider-impl.ts

```diff
@@ -45,41 +49,36 @@ export class TraceServerUrlProviderImpl implements TraceServerUrlProvider, Front
         return this._onDidChangeTraceServerUrlEmitter.event;
     }
 
-    constructor(
-        @inject(EnvVariablesServer) protected environment: EnvVariablesServer,
-        @inject(TracePreferences) protected tracePreferences: TracePreferences,
-        @inject(TraceServerConfigService) protected traceServerConfigService: TraceServerConfigService,
-        @inject(MessageService) protected messageService: MessageService
-    ) {
+    constructor(@inject(TraceServerConfigService) protected traceServerConfigService: TraceServerConfigService) {
         this._traceServerUrlPromise = new Promise(resolve => {
             const self = this.onDidChangeTraceServerUrl(url => {
                 self.dispose();
                 resolve(url);
             });
         });
         // Get the URL template from the remote environment.
-        this.environment.getValue('TRACE_SERVER_URL').then(variable => {
-            const url = variable?.value;
-            this._traceServerUrlTemplate = url ? this.normalizeUrl(url) : TRACE_SERVER_DEFAULT_URL;
-            this.updateTraceServerUrl();
-        });
-        // Get the configurable port from Theia's preferences.
-        this.tracePreferences.ready.then(() => {
-            this._traceServerPort = this.tracePreferences[TRACE_PORT];
-            this.updateTraceServerUrl();
-            this.tracePreferences.onPreferenceChanged(async event => {
-                if (event.preferenceName === TRACE_PORT) {
-                    this._traceServerPort = event.newValue;
-                    this.updateTraceServerUrl();
-                    try {
-                        await this.traceServerConfigService.stopTraceServer();
-                        this.messageService.info(`Trace server disconnected on port: ${event.oldValue}.`);
-                    } catch (_) {
-                        // Do not show the error incase the user tries to modify the port before starting a server
-                    }
+        const variable = process.env['TRACE_SERVER_URL'];
+        this._traceServerUrlTemplate = variable ? this.normalizeUrl(variable) : TRACE_SERVER_DEFAULT_URL;
+        this.updateTraceServerUrl();
+    }
+
+    async onPortPreferenceChanged(
+        newPort: number | undefined,
+        oldValue?: number,
+        preferenceChanged = false
+    ): Promise<void> {
+        this._traceServerPort = newPort;
```

Comment:
Changing the port doesn't seem to be propagate properly to the TspClient and messages are not successful.

To reproduce:
- Command: `Start Trace Server`
- Open Settings -> Trace Viewer
- Change Port to 8081 (which will stop the trace server)
- Change "Trace Server: Arguments" to -vmargs -Dtraceserver.port=8081
- Command `Start Trace Server`
- There are error messages: "ERROR Error: Unsuccessful health check: Service Unavailable status: undefined"

Note: the server process is actually started (do "ps -aef  | grep trace | grep server")
Note: heath check is executed after the trace sever is spawned

I did debug a bit and noticed that the http-tsp-client in the backend doesn't use the correct port.

Reply:
I will need to dig a little bit more into this to find out why this happens. Just for my understanding:
The Port preference is simply there for the requests to the server, while the params will be used for the startup of the Trace Server (when startet from Theia), right?

And changing the Port should kill all servers running on the old Port?

Reply:
This behavior should now be fixed with the latest commit. I also rebased on master.

Reply:
Yes, it's fixed.